gammapy/datasets/tests/test_spectrum.py

```diff
@@ -629,6 +633,20 @@ def test_spectrum_dataset_onoff_fits_io(self, tmp_path):
         assert d1.exposure == self.dataset.exposure
         assert_allclose(d1.counts_off.data, self.dataset.counts_off.data)
 
+    def test_spectrum_dataset_onoff_default_format(self, tmp_path):
+        self.dataset.write(tmp_path / "test.fits")
+        d1 = SpectrumDatasetOnOff.read(tmp_path / "test.fits")
+        assert isinstance(d1.counts.geom, RegionGeom)
+        assert d1.exposure == self.dataset.exposure
+        assert_allclose(d1.counts_off.data, self.dataset.counts_off.data)
+
+    def test_spectrum_dataset_onoff_fallback_from_gadf(self, tmp_path):
+        self.dataset.write(tmp_path / "test.fits", format="ogip")
+        d1 = SpectrumDatasetOnOff.read(tmp_path / "test.fits")
+        assert isinstance(d1.counts.geom, RegionGeom)
+        assert d1.exposure == self.dataset.exposure
+        assert_allclose(d1.counts_off.data, self.dataset.counts_off.data)
+
```

Comment:
This misses a test of the format identification with the proper exception raise.

Reply:
I believe it is implemented now 

> This misses a test of the format identification with the proper exception raise.